test/unit/L2/l2Migrator.test.ts

```diff
@@ -146,37 +158,124 @@ describe('L2Migrator', function() {
       const ticketBrokerAddr = await l2Migrator.ticketBrokerAddr();
       expect(ticketBrokerAddr).to.equal(mockTicketBrokerEOA.address);
+      const merkleSnapshotAddr = await l2Migrator.merkleSnapshotAddr();
+      expect(merkleSnapshotAddr).to.equal(mockMerkleSnapshotEOA.address);
     });
   });
 
   describe('initialize', () => {
     describe('when caller not controller owner', async () => {
       it('should fail to set addresses', async () => {
         const addr = ethers.constants.AddressZero;
-        const tx = l2Migrator
-            .connect(owner)
-            .initialize(addr, addr, addr, addr, addr);
+        const tx = l2Migrator.connect(owner).initialize(addr, addr);
         await expect(tx).to.be.revertedWith('caller must be Controller owner');
       });
     });
 
     describe('when caller is controller owner', async () => {
       it('should set addresses', async () => {
         const addr = ethers.constants.AddressZero;
-        await l2Migrator
-            .connect(admin)
-            .initialize(addr, addr, addr, addr, addr);
+        await l2Migrator.connect(admin).initialize(addr, addr);
 
         const l1MigratorAddr = await l2Migrator.l1Migrator();
         expect(l1MigratorAddr).to.equal(addr);
         const delegatorPoolImpl = await l2Migrator.delegatorPoolImpl();
         expect(delegatorPoolImpl).to.equal(addr);
         const bondingManagerAddr = await l2Migrator.bondingManagerAddr();
-        expect(bondingManagerAddr).to.equal(addr);
+        expect(bondingManagerAddr).to.equal(mockBondingManagerEOA.address);
         const ticketBrokerAddr = await l2Migrator.ticketBrokerAddr();
-        expect(ticketBrokerAddr).to.equal(addr);
+        expect(ticketBrokerAddr).to.equal(mockTicketBrokerEOA.address);
         const merkleSnapshotAddr = await l2Migrator.merkleSnapshotAddr();
-        expect(merkleSnapshotAddr).to.equal(addr);
+        expect(merkleSnapshotAddr).to.equal(mockMerkleSnapshotEOA.address);
+      });
+    });
+  });
+
+  describe('syncControllerContracts', () => {
+    const bondingManagerID = ethers.utils.solidityKeccak256(
+        ['string'],
+        ['BondingManager'],
+    );
+    const ticketBrokerID = ethers.utils.solidityKeccak256(
+        ['string'],
+        ['TicketBroker'],
+    );
+    const merkleSnapshotID = ethers.utils.solidityKeccak256(
+        ['string'],
+        ['MerkleSnapshot'],
+    );
+
+    describe('addresses do not change', () => {
+      it('should set addresses', async () => {
+        const tx = l2Migrator.syncControllerContracts();
+        await expect(tx).to.not.emit(l2Migrator, 'ProtocolContractUpdate');
+      });
+    });
+
+    describe('only bondingManager Changes', () => {
+      it('should set addresses', async () => {
+        controllerMock.getContract
+            .whenCalledWith(bondingManagerID)
+            .returns(ethers.constants.AddressZero);
+
+        const tx = await l2Migrator.syncControllerContracts();
+
+        await expect(tx)
```

Comment:
For completeness should these tests also check the following?

- There is no event emitted for the other contracts
- The stored address for the relevant contract is updated